src/components/inputs/CPriceInput.vue

```diff
@@ -0,0 +1,158 @@
+    color="primary"
+    @input="onInputChange"
+  ></v-text-field>
+</template>
+
+<script setup lang="ts">
+import { mdiCurrencyEur } from "@mdi/js";
+import { ref, watch } from "vue";
+
+/**
+ * The currently entered price.
+ */
+const model = defineModel<number | undefined>({ required: true });
+
+/**
+ * The model for the input field.
+ *
+ * This field is just a ref and will only be updated from the field itself and the watcher of the `model`.
+ *
+ * A computed ref is not used because the parse functions will automatically remove any disallowed chars on change.
+ * This causes the `model`s value to not change and therefore, the computed not to be updated, which leads to the problem
+ * that disallowed chars are not removed anymore.
+ */
+const fieldValue = ref<string | undefined>();
+
+/**
+ * Watches the model to set the newest value to `fieldValue`.
+ *
+ * This one will only use `formatPrice` to also support database formats like `12` to be parsed to `12,00` instead of `0,12`.
+ */
+watch(
+  model,
+  () => {
+    // Update field value to fit format
+    fieldValue.value = model.value
+      ? model.value.toFixed(2).toString().replace(".", ",")
+      : undefined;
+  },
+  { immediate: true },
+);
+
+/**
+ * This component builds an input field for a price with simplified typing experience.
+ *
+ * The format of the price will always be `x,xx`.
+ */
+withDefaults(
+  defineProps<{
+    /**
+     * The label of this field.
+     */
+    label: string;
+
+    /**
+     * The variant of this field.
+     */
+    variant?:
+      | "underlined"
+      | "outlined"
+      | "filled"
+      | "solo"
+      | "solo-inverted"
+      | "solo-filled"
+      | "plain";
+
+    /**
+     * Whether the input fields should be disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * The rules to use to validate the password.
+     */
+    rules: ((password: string) => true | string)[];
+  }>(),
+  { variant: "outlined" },
+);
+
+/**
+ * Parses the given string value into a price with the format `x,xx`.
+ * The formatted price can easily be parsed into a float value with `stringToFloat`.
+ *
+ * This function is designed to support typing the price from start to end without caring about the comma.
+ *
+ * @param value the value to parse
+ */
+function parsePrice(value?: string): string | undefined {
+  if (!value) return undefined;
+
+  // Remove `,` and `.`
+  value = value.replace(",", "");
+  value = value.replace(".", "");
+
+  // Parse float to remove leading 0s
+  const floatVal: number = parseFloat(value);
+
+  if (isNaN(floatVal) || floatVal === 0) {
+    return undefined;
+  }
+
+  const stringVal = floatVal.toString();
+
+  // Add decimal divider
+  if (stringVal.length === 1) {
+    return "0,0" + stringVal;
+  } else if (stringVal.length === 2) {
+    return "0," + stringVal;
+  } else {
+    return (
+      stringVal.slice(0, -2) +
+      "," +
+      (stringVal.length > 1 ? stringVal.slice(-2) : "0" + stringVal)
+    );
+  }
```

Comment:
why don't you use something linke `toLocaleString`?

Reply:
because `toLocaleString` does not format the number `12` into `0,12`

Reply:
then divide by 100 beforehand 😅

Reply:
because this function will be called every time the input changes. This would cause the a number 12,00 to be parsed to 0,12, typing a 1 again will result in a 0,121, parsed to 1,21 divided by 100 => 0,0121, this does not work.

I am sure, this can be simplified but I am fine with it working now. We can have a look at this later on

Reply:
Thats not true, you remove all `.` and `,` above before parsing it to a float

Reply:
simplified it